test/test_other.py

```diff
@@ -5942,11 +5942,19 @@ def test_force_stdlibs(self):
     self.do_runf('hello_world.c', emcc_args=['-sWASM_BIGINT'])
 
   @crossplatform
-  @also_with_env_modify({'gb_locale': {'LC_ALL': 'en_GB'}, 'long_tz': {'TZ': 'Asia/Kathmandu'}})
   def test_strftime_zZ(self):
-    if os.environ.get('LC_ALL') == 'en_GB' and MACOS:
+    if MACOS:
       self.skipTest('setting LC_ALL is not compatible with macOS python')
-    self.do_runf('other/test_strftime_zZ.c', 'ok!')
+
+    tz_lang_envs = [
+        {"LC_ALL": "en_GB", "TZ": "Europe/London"},
+        {"LC_ALL": "th_TH", "TZ": "Asia/Bangkok"},
+        {"LC_ALL": "ar-AE", "TZ": "United Arab Emirates"},
+    ]
+
+    for tz_lang_env in tz_lang_envs:
+        with env_modify(tz_lang_env):
+            self.do_runf('other/test_strftime_zZ.c', 'ok!')
```

Comment:
We use space indentation for python code.

Reply:
I wonder if we should update this test to check for the specific output.

Right now it it just checks for `ok`, it looks like the test also print `printf("The current timezone is: %s\n", timezone);`.

Reply:
> We use space indentation for python code.

Somehow, my text editor wanted to go wild 😅 🤦. Fixed :)  

> I wonder if we should update this test to check for the specific output.

yes that makes total sense, updated the test to check for correct TZ extraction and added a new case to test both positive and negative offset : https://github.com/emscripten-core/emscripten/pull/22250/commits/d72b05b463cd71d3cf644f965cd06ab584243ffc